src/Nethermind/Nethermind.Merge.Plugin/MergeGossipPolicy.cs

```diff
@@ -16,31 +16,42 @@
 // 
 
 using System;
-using Nethermind.Blockchain;
 using Nethermind.Consensus;
-using Nethermind.Core.Crypto;
+using Nethermind.Core;
 
 namespace Nethermind.Merge.Plugin
 {
     public class MergeGossipPolicy : IGossipPolicy
     {
         private readonly IGossipPolicy _preMergeGossipPolicy;
         private readonly IPoSSwitcher _poSSwitcher;
-        private readonly IManualBlockFinalizationManager _blockFinalizationManager;
+        private readonly IMergeConfig _mergeConfig;
 
         public MergeGossipPolicy(
-            IGossipPolicy? apiGossipPolicy, 
-            IPoSSwitcher? poSSwitcher, 
-            IManualBlockFinalizationManager blockFinalizationManager)
+            IGossipPolicy? apiGossipPolicy,
+            IPoSSwitcher? poSSwitcher,
+            IMergeConfig? mergeConfig)
         {
             _preMergeGossipPolicy = apiGossipPolicy ?? throw new ArgumentNullException(nameof(apiGossipPolicy));
             _poSSwitcher = poSSwitcher ?? throw new ArgumentNullException(nameof(poSSwitcher));
-            _blockFinalizationManager = blockFinalizationManager;
+            _mergeConfig = mergeConfig ?? throw new ArgumentNullException(nameof(mergeConfig));
         }
 
-        public bool ShouldGossipBlocks => !_poSSwitcher.HasEverReachedTerminalBlock() && _preMergeGossipPolicy.ShouldGossipBlocks;
+        // According to spec (https://github.com/ethereum/EIPs/blob/d896145678bd65d3eafd8749690c1b5228875c39/EIPS/eip-3675.md#network)
+        // We SHOULD NOT advertise the descendant of any terminal PoW block.
+        // In our approach we CAN gossip blocks if we didn't received FIRST_FINALIZED_BLOCK yet...
+        public bool CanGossipBlocks => !_poSSwitcher.TransitionFinished && _preMergeGossipPolicy.CanGossipBlocks;
 
-        public bool ShouldDisconnectGossipingNodes => _blockFinalizationManager.LastFinalizedHash != Keccak.Zero && _preMergeGossipPolicy.ShouldDisconnectGossipingNodes;
+        //  ...and gossipping policy will be decided according to the header difficulty.
+        public bool ShouldGossipBlock(BlockHeader header) => !_poSSwitcher.GetBlockConsensusInfo(header).IsPostMerge;
```

Comment:
add condition with TransitionFinished or _mergeConfig.FinalTotalDifficulty?

Reply:
not needed here. It's used only once, only if CanGossip is true, so TransitionFinished is already checked
https://github.com/NethermindEth/nethermind/blob/c10515bbd76763e443d64e8750094393969b23ba/src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V62/Eth62ProtocolHandler.cs#L280